addon/lint/lint.js

```diff
@@ -187,7 +187,7 @@
   CodeMirror.defineOption("lint", false, function(cm, val, old) {
     if (old && old != CodeMirror.Init) {
       clearMarks(cm);
-      if (!state.options.lintOnChange !== false)
+      if (cm.state.lint.options.lintOnChange === undefined || cm.state.lint.options.lintOnChange === true)
```

Comment:
No, I don't think you are right. `true !== false` is true. With the `!== false` condition you only turn of the change event handler when `lintOnChange` is explicitly set to `false`, which is what we want.